modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js

```diff
@@ -141,8 +173,8 @@ export default function Navigation({
 						canonicalURL={canonicalURL}
 						chartDataProviders={
 							endpoints.analyticsReportsHistoricalReadsURL
-								? [getHistoricalViews, getHistoricalReads]
-								: [getHistoricalViews]
+								? [handleHistoricalViews, handleHistoricalReads]
+								: [handleHistoricalViews]
```

Comment:
In terms of reviewing a big refactor like this, I am just looking at an isolated PR to see if anything looks off, but not really questioning the big-picture design of it — that's something we can look at at the end, if you like.

Only question I have here looking at an isolated snippet like this is, are we `catch()`-ing possible promise rejections in the place where we handle this prop? (I expect we are, because I gather the old API was also dealing with promises, but just thought I'd ask.)

Reply:
Sure! I would like to take a look at the end of the refactor to review the design/patters/problems of it 👍  I think it's a good idea and I can take advantage of that and learn for future implementations.

About the `catch()`-ing, for historical data (historical views and reads) you may remember our `allSettled` method (https://github.com/liferay/liferay-portal/blob/master/modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js#L457-L469) that we used to wait until both promises end to show the historical data. In that method, we have a then/catch logic and in the `then` part of the `allSettled` we check if the promise is `fullfilled`. If not, we add like an empty set of data to show the legend with the empty chart (https://github.com/liferay/liferay-portal/blob/master/modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js#L198-L222).